src/sentry/api/invite_helper.py

```diff
@@ -226,7 +226,7 @@ def _needs_email_verification(self) -> bool:
 
         user = self.request.user
         primary_email_is_verified = (
-            isinstance(user, User) and UserEmail.objects.get_primary_email(user).is_verified
+            isinstance(user, BaseUser) and UserEmail.objects.get_primary_email(user).is_verified
```

Comment:
Hmm, @mikejihbe Is this change going to truly be necessary?  I think we should be able to revert this, actually.

Reply:
it might not be. It seemed safer to over-apply the BaseUser check than to under apply it, so that's what I did.